worlds/civ_6/__init__.py

```diff
@@ -0,0 +1,205 @@
+
+    def get_filler_item_name(self) -> str:
+        return get_random_filler_by_rarity(self, FillerItemRarity.COMMON).name
+
+    def create_regions(self) -> None:
+        create_regions(self, self.options, self.player)
+
+    def set_rules(self) -> None:
+        if self.options.boostsanity:
+            create_boost_rules(self)
+
+    def create_item(self, name: str) -> Item:
+        item: CivVIItemData = self.item_table[name]
+        classification = item.classification
+        if self.options.boostsanity:
+            if item.civ_name in BOOSTSANITY_PROGRESSION_ITEMS:
+                classification = ItemClassification.progression
+
+        return CivVIItem(item, self.player, classification)
+
+    def create_items(self) -> None:
+        data = get_era_required_items_data()
+        early_items = data[EraType.ERA_ANCIENT.value]
+        early_locations = [location for location in self.location_table.values() if location.era_type == EraType.ERA_ANCIENT.value]
+        for item_name, item_data in self.item_table.items():
+            # These item types are handled individually
+            if item_data.item_type in [CivVICheckType.PROGRESSIVE_DISTRICT, CivVICheckType.ERA, CivVICheckType.GOODY]:
+                continue
+
+            # If we're using progressive districts, we need to check if we need to create a different item instead
+            item_to_create = item_name
+            item: CivVIItemData = self.item_table[item_name]
+            if self.options.progression_style != "none":
+                if item.progression_name:
+                    item_to_create = self.item_table[item.progression_name].name
+
+            self.multiworld.itempool += [self.create_item(
+                item_to_create)]
+            if item.civ_name in early_items:
+                self.multiworld.early_items[self.player][item_to_create] = 1
+            elif self.item_table[item_name].era in [EraType.ERA_ATOMIC, EraType.ERA_INFORMATION, EraType.ERA_FUTURE]:
+                for location in early_locations:
+                    found_location = None
+                    try:
+                        found_location = self.get_location(location.name)
+                        forbid_item(found_location, item_to_create, self.player)
+                    except KeyError:
+                        pass
+
+        # Era items
+        if self.options.progression_style == "eras_and_districts":
+            # Add one less than the total number of eras (start in ancient, don't need to find it)
+            for era in EraType:
+                if era.value == "ERA_ANCIENT":
+                    continue
+                progressive_era_item = self.item_table.get("Progressive Era")
+                if progressive_era_item:
+                    self.multiworld.itempool += [self.create_item(progressive_era_item.name)]
+                else:
+                    logging.error("Progressive Era item not found in item_table.")

```

Comment:
Do you actually expect to hit this error? You might want to change this to an assert instead